src/main/kotlin/org/rust/ide/highlight/RsRainbowVisitor.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.highlight
+
+import com.intellij.codeInsight.daemon.RainbowVisitor
+import com.intellij.codeInsight.daemon.impl.HighlightVisitor
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.descendantsOfType
+import org.rust.lang.core.psi.ext.parentOfType
+
+class RsRainbowVisitor : RainbowVisitor() {
+    override fun suitableForFile(file: PsiFile): Boolean = file is RsFile
+
+    override fun clone(): HighlightVisitor = RsRainbowVisitor()
+
+    override fun visit(element: PsiElement) {
```

Comment:
Ah, this is another kind of visitor altogether... Sorry for causing confusion then! Let's merge it, and I'll try to do some cleanups myself!